### ISSUE TITLE
PS2Net not working

### ISSUE THREAD

**Issue by Windows10User2** [israpps/wLaunchELF_ISR#79]
### Checks

- [ ] I have checked existing [__Issue reports__](https://github.com/israpps/wLaunchELF_ISR/issues) for duplicates and found none
- [ ] I downloaded wLaunchELF from this repository or the [official download link](https://israpps.github.io/20210113_wLaunchELF_isr/)

### Console model

SCPH-70004

### wLaunchELF commit code

e31a9a3

### Describe the issue

PS2Net doesn't work in the BOOT-EXFAT-DS34.ELF version. Even though I insert the right IP address on FileZilla Client, it can't access the PS2.

**Comment by israpps** [comment]
> SCPH-90008 > OpenTuna > wLE_ISR > PS2Net > Total Commander FTP > Connect call failed
> 
> SCPH-90008 > OpenTuna > wLE_kHn > PS2Net > Total Commander FTP > Works well
> 
> @israpps, this issue also exists in the official wLE. More details [here](https://github.com/ps2homebrew/wLaunchELF/issues/123#issuecomment-2241471995).

Try here
https://www.mediafire.com/file/p1jc0sn8lhmryfw/WLE-REVERT-CDVD.zip/file

**Comment by Windows10User2** [comment]
> SCPH-90008 > OpenTuna > wLE_ISR > PS2Net > Total Commander FTP > Connect call failed
> 
> SCPH-90008 > OpenTuna > wLE_kHn > PS2Net > Total Commander FTP > Works well
> 
> @israpps, this issue also exists in the official wLE. More details [here](https://github.com/ps2homebrew/wLaunchELF/issues/123#issuecomment-2241471995).

It didn't exist in wLE_ISR's 945d787 commit and you told me [here](https://github.com/ps2homebrew/wLaunchELF/issues/121#issuecomment-1933203234) the issue had been fixed in a previous commit in the official wLE.

**Comment by Windows10User2** [comment]
> > SCPH-90008 > OpenTuna > wLE_ISR > PS2Net > Total Commander FTP > Connect call failed
> > SCPH-90008 > OpenTuna > wLE_kHn > PS2Net > Total Commander FTP > Works well
> > @israpps, this issue also exists in the official wLE. More details [here](https://github.com/ps2homebrew/wLaunchELF/issues/123#issuecomment-2241471995).
> 
> Try here https://www.mediafire.com/file/p1jc0sn8lhmryfw/WLE-REVERT-CDVD.zip/file

Will that fix the FTP problem and is it a wLE official version or a wLE_ISR one? Also, will [this](https://github.com/israpps/wLaunchELF_ISR/issues/62) ever be fixed?

**Comment by israpps** [comment]
> > > SCPH-90008 > OpenTuna > wLE_ISR > PS2Net > Total Commander FTP > Connect call failed
> > > SCPH-90008 > OpenTuna > wLE_kHn > PS2Net > Total Commander FTP > Works well
> > > @israpps, this issue also exists in the official wLE. More details [here](https://github.com/ps2homebrew/wLaunchELF/issues/123#issuecomment-2241471995).
> > 
> > 
> > Try here https://www.mediafire.com/file/p1jc0sn8lhmryfw/WLE-REVERT-CDVD.zip/file
> 
> Will that fix the FTP problem and is it a wLE official version or a wLE_ISR one? Also, will [this](https://github.com/israpps/wLaunchELF_ISR/issues/62) ever be fixed?

#62 is not something I can fix easily. that limitation is on `MX4SIO_BD.IRX` driver itself...

**Comment by Windows10User2** [comment]
> > > > SCPH-90008 > OpenTuna > wLE_ISR > PS2Net > Total Commander FTP > Connect call failed
> > > > SCPH-90008 > OpenTuna > wLE_kHn > PS2Net > Total Commander FTP > Works well
> > > > @israpps, this issue also exists in the official wLE. More details [here](https://github.com/ps2homebrew/wLaunchELF/issues/123#issuecomment-2241471995).
> > > 
> > > 
> > > Try here https://www.mediafire.com/file/p1jc0sn8lhmryfw/WLE-REVERT-CDVD.zip/file
> > 
> > 
> > Will that fix the FTP problem and is it a wLE official version or a wLE_ISR one? Also, will [this](https://github.com/israpps/wLaunchELF_ISR/issues/62) ever be fixed?
> 
> #62 is not something I can fix easily. that limitation is on `MX4SIO_BD.IRX` driver itself...

What about the FTP problem?

**Comment by israpps** [comment]
> What about the FTP problem?

Very hard to debug without a devkit. will check it later

**Comment by Windows10User2** [comment]
> > What about the FTP problem?
> 
> Very hard to debug without a devkit. will check it later

I find it odd a previous commit didn't have it.

**Comment by israpps** [comment]
> > > What about the FTP problem?
> > 
> > 
> > Very hard to debug without a devkit. will check it later
> 
> I find it odd a previous commit didn't have it.

Network was not touched.

Try the test build I sent here before

**Comment by Windows10User2** [comment]
> > > > What about the FTP problem?
> > > 
> > > 
> > > Very hard to debug without a devkit. will check it later
> > 
> > 
> > I find it odd a previous commit didn't have it.
> 
> Network was not touched.
> 
> Try the test build I sent here before

Then I don't know why I didn't have this issue and now I do.

**Comment by Windows10User2** [comment]
> as I said twice already
> 
> > Try here https://www.mediafire.com/file/p1jc0sn8lhmryfw/WLE-REVERT-CDVD.zip/file
> 
> and tell me if it works...

I just tried that BOOT-EXFAT-DS34 version and it works. Will this commit be released?

**Comment by israpps** [comment]
WTF

This shit is insane

**Comment by Windows10User2** [comment]
> WTF
> 
> This shit is insane

What do you mean?

> [@israpps](https://github.com/israpps), is it possible to fix [this](https://github.com/ps2homebrew/wLaunchELF/issues/123), [this](https://github.com/ps2homebrew/wLaunchELF/issues/122) and [this](https://github.com/ps2homebrew/wLaunchELF/issues/121#issuecomment-1918207666) issue in the official wLE?

What do the latter two issues have to do with [mine](https://github.com/israpps/wLaunchELF_ISR/issues/79#issue-2409484535)?

**Comment by israpps** [comment]
Forget of fixing ps2net on official wLaunchELF.

What I did that caused this issue was using latest CDFS.IRX to achieve freedvdboot compatibility.

They will not fall back to the legacy one.

Not sure how the hell a driver related to extending access to disc drive related features could impact negatively the network related stuff.


Not to mention, no one will probably waste time on tracing that issue.
Like everything related to the PS1 CPU, debugging is hard AF

**Comment by Windows10User2** [comment]
> Forget of fixing ps2net on official wLaunchELF.
> 
> What I did that caused this issue was using latest CDFS.IRX to achieve freedvdboot compatibility.
> 
> They will not fall back to the legacy one.
> 
> Not sure how the hell a driver related to extending access to disc drive related features could impact negatively the network related stuff.
> 
> Not to mention, no one will probably waste time on tracing that issue. Like everything related to the PS1 CPU, debugging is hard AF

And does the wLE version that comes with FDVDB have this PS2Net issue? I thought only the latest commits did. If so, you could discard FDVDB in the latest commits and fix the PS2Net problem and let users run FDVDB with a working PS2Net even if its wLE is outdated.

**Comment by Windows10User2** [comment]
After clicking "mc0:/", wLE_isr latest automated build's BOOT-EXFAT-DS34 opens it and sometimes immediately exits out of it as if I had pressed the "Triangle" button. I have to restart the PS2 when I have this problem and run wLE_isr latest automated build's BOOT-EXFAT-DS34 again. I don't know if it's Brook Super Converter's or DualShock 3's fault.

**Comment by israpps** [comment]
> When running wLE_ISR on the PS2 and clicking on a folder ("mc0:/", for instance), it often opens it and immediately turns back, 

That issue is already known and I don't know what's causing the issue

**Comment by Windows10User2** [comment]
> > When running wLE_ISR on the PS2 and clicking on a folder ("mc0:/", for instance), it often opens it and immediately turns back,
> 
> That issue is already known and I don't know what's causing the issue

So, it's not Brook Super Converter's or DualShock 3's fault? Does it happen on the official wLE as well or only in any wLE_ISR version?

**Comment by israpps** [comment]
> > > When running wLE_ISR on the PS2 and clicking on a folder ("mc0:/", for instance), it often opens it and immediately turns back,
> > 
> > 
> > That issue is already known and I don't know what's causing the issue
> 
> So it's not Brook Super Converter's or DualShock 3's fault? Does it happen on the official wLE as well or only in any wLE_ISR version?

It happens on wle Isr only, and I don't know why

Or why it only happens on mc0

**Comment by Windows10User2** [comment]
> > > > When running wLE_ISR on the PS2 and clicking on a folder ("mc0:/", for instance), it often opens it and immediately turns back,
> > > 
> > > 
> > > That issue is already known and I don't know what's causing the issue
> > 
> > 
> > So it's not Brook Super Converter's or DualShock 3's fault? Does it happen on the official wLE as well or only in any wLE_ISR version?
> 
> It happens on wle Isr only, and I don't know why
> 
> Or why it only happens on mc0

So, it only happens on wLE_ISR and only on mc0, not on wLE_ISR and mc1 or wLE and mc0?

**Comment by israpps** [comment]
> @israpps, what about [this](https://github.com/ps2homebrew/wLaunchELF/issues/122) and [this](https://github.com/ps2homebrew/wLaunchELF/issues/121#issuecomment-1918207666) issue? They didn't exist before, they appeared in [Latest development build](https://github.com/ps2homebrew/wLaunchELF/releases/tag/latest).

Impossible to track them down.

5 years passed between the build in Wich wLaunchELF Isr is based and latest wLaunchELF.

Not only the changes to wLaunchELF itself can be responsible, the PS2SDK (toolchain, SDK, etc) changed a lot and also have influence.

**Comment by syboxez** [comment]
This issue still exists in the latest commit. I built it myself using `ETH=1 EXFAT=1 DS34=0 MMCE=1 XFROM=0`
It also exists in the release builds

**Comment by Windows10User2** [comment]
> [@israpps](https://github.com/israpps), what about [this](https://github.com/ps2homebrew/wLaunchELF/issues/122) and [this](https://github.com/ps2homebrew/wLaunchELF/issues/121#issuecomment-1918207666) issue? They didn't exist before, they appeared in [Latest development build](https://github.com/ps2homebrew/wLaunchELF/releases/tag/latest).

Again? What do those issues have to do with [mine](https://github.com/israpps/wLaunchELF_ISR/issues/79#issue-2409484535)? Anyway, when trying to connect to the PS2 by FTP using FileZilla Client in Windows 11 and wLE_isr latest automated build's BOOT-EXFAT-MMCE's PS2Net, FileZilla Client no longer freezes at "Loading NetFS and FTP Server Modules..." but now it says:

"Status: Connection established, waiting for welcome message...
Status: Plain FTP is insecure. Please switch to FTP over TLS.
Status: Server does not support non-ASCII characters.
Status: Logged in
Status: Retrieving directory listing...
Status: Directory listing of "/" successful"

Below "Remote site:", Filezilla Client only shows a mount point called "/" and below "Filename" it only shows one called "..". I've tried other wLE and wLE_isr builds and FTP programs (including Windows 11's File Explorer) and they don't work too. This is odd since it was working before in the previous wLE_isr automated build so maybe the latest (which has the https://github.com/israpps/wLaunchELF_ISR/commit/74550d86c08b44ee00470ea9e39c0b7f17a12bef commit) is the culprit. This also happens in wLE_isr latest automated build's BOOT-EXFAT-DS34.

**Comment by Windows10User2** [comment]
> This issue still exists in the latest commit. I built it myself using `ETH=1 EXFAT=1 DS34=0 MMCE=1 XFROM=0` It also exists in the release builds

It's a different issue now, like I've explained [here](https://github.com/israpps/wLaunchELF_ISR/issues/79#issuecomment-4234006780) and [here](https://github.com/israpps/wLaunchELF_ISR/issues/113#issue-4195276789).

EDIT: I tried wLE 4.43a 41e4ebe and it doesn't have this problem so wLE_isr's latest automated build (which has the 74550d8 commit) really is the culprit. Anyway, I think I'll be using USB devices from now on in wLE_isr latest automated build's BOOT-EXFAT-MMCE when wanting to copy files from there to the MCs or just run them from the USB devices until a new wLE_isr automated build without this problem is out (if it will ever be).
